bookkeeper-server/build.gradle

```diff
@@ -99,4 +99,7 @@ test.doFirst {
         .resolvedArtifacts.find { it.name == 'junit-foundation' }
     jvmArgs("-Djunit.timeout.test=600000", "-Djunit.max.retry=3",
             "-Djava.net.preferIPv4Stack=true", "-Dio.netty.leakDetection.level=paranoid")
+    if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
+        jvmArgs("--add-opens", "java.base/jdk.internal.loader=ALL-UNNAMED")
```

Comment:
What problem does this PR solve? Or in other words what this jvmArgs does and lacking this will result in what problem?

Reply:
Another thins to note is. For maven `mvn clean package -B -nsu -DskipBookKeeperServerTests -Dorg.slf4j.simpleLogger.defaultLogLevel=INFO ` therefore for the gradle we are not running bookkeeper-server test on Java 11 as of now.

Reply:
I think we should make sure the tests are passing even without that flag, otherwise we might end up in a scenario where the flag is needed for running the bookie or bk client.

Reply:
I agree with @merlimat we should find another way of doing this.

@pkumar-singh this is my error https://github.com/apache/bookkeeper/issues/2872 that this PR is trying to solve

Reply:
@merlimat @eolivelli I think it only affects the tests. I found a similar fix in pulsar https://github.com/apache/pulsar/pull/9893